### PR TITLE
NAS-119346 / 23.10 / Add ddns-updater to `community`

### PR DESCRIPTION
**Pull request by stavros-k**
All upstream supported providers have been added, 
App will run rootless (configurable).
On all CI runs probes are disable, because healthcheck command of the container will fail without valid API tokens etc.
But we still run those tests as the container will exit if configuration is invalid. (For example there is a directory `failing-tests` with 1 provider there that panics and causes test to fail, related fix is pushed upstream waiting for review/merge).

Base is made extendable, which makes adding another provider very easy with a few lines. 

---

Along the way few typos in docs found and pushed upstream fixes + couple of bugs.

Related fixes pushed upstream:
https://github.com/qdm12/ddns-updater/pull/462
https://github.com/qdm12/ddns-updater/pull/473

Related doc updates pushed upstream:
https://github.com/qdm12/ddns-updater/pull/463
https://github.com/qdm12/ddns-updater/pull/465
https://github.com/qdm12/ddns-updater/pull/466
https://github.com/qdm12/ddns-updater/pull/467
https://github.com/qdm12/ddns-updater/pull/468
https://github.com/qdm12/ddns-updater/pull/469
https://github.com/qdm12/ddns-updater/pull/470
https://github.com/qdm12/ddns-updater/pull/471
https://github.com/qdm12/ddns-updater/pull/472

Related feature requests:
https://github.com/qdm12/ddns-updater/issues/464

### CODE REVIEW

File: library/ix-dev/community/ddns-updater/values.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+image:
+  repository: qmcgaw/ddns-updater
+  pullPolicy: IfNotPresent
+  tag: latest
```
[the remark this answers is not in the document]
Comment (stavros-k) [reply library/ix-dev/community/ddns-updater/values.yaml:4]: https://github.com/qdm12/ddns-updater/issues/464

[the remark this answers is not in the document]
Comment (stavros-k) [reply library/ix-dev/community/ddns-updater/values.yaml:4]: Strategy is there in hope upstream will do a release "soon" before this is merged,
If thats not the case, strategy will be removed

**Review by Qubad786 — APPROVED** [review]
Let's remove the strategy in that case ✅

File: library/ix-dev/community/ddns-updater/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (qdm12) [reply library/ix-dev/community/ddns-updater/values.yaml:4]: Released v2.5.0 sorry guys this will be more regular from now on.

**Pull request merged**